python/ray/data/_internal/datasource/parquet_datasource.py

```diff
@@ -33,6 +33,7 @@
     get_generic_metadata_provider,
 )
 from ray.data.datasource.datasource import ReadTask
+from ray.data.datasource.file_based_datasource import _validate_shuffle_arg
```

Comment:
To avoid coupling `file_based_datasource` and `parquet_datasource`, I'd rather duplicate duplicate this function in `parquet_datasource`

Reply:
Sounds good! And just to confirm, should these validations be happening in the `Datasource` layer or at the `read_xyz` layer?